client/question/services/SessionHistoryServiceSpec.js

```diff
@@ -0,0 +1,126 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the SessionHistoryService.
+ */
+
+describe('SessionHistoryService', function() {
+  var SessionHistoryService;
+  var FeedbackParagraphObjectFactory;
+  var DURATION_MSEC_WAIT_FOR_FEEDBACK;
+  var $timeout;
+
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector, _$timeout_) {
+    SessionHistoryService = $injector.get('SessionHistoryService');
+    FeedbackParagraphObjectFactory = $injector.get(
+      'FeedbackParagraphObjectFactory');
+    DURATION_MSEC_WAIT_FOR_FEEDBACK = $injector.get(
+      'DURATION_MSEC_WAIT_FOR_FEEDBACK');
+    $timeout = _$timeout_;
+  }));
+
+  describe('core behaviour', function() {
```

Comment:
there's a huge opportunity for me to get american sassy and correct this to "behavior" but I'll allouw it